packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts

```diff
@@ -36,6 +45,32 @@ export class CCreateLoanScheme extends ComposableBuffer<CreateLoanScheme> {
   }
 }
 
+/**
+ * Composable DestroyLoanScheme, C stands for Composable.
+ * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+ */
+export class CDestroyLoanScheme extends ComposableBuffer<DestroyLoanScheme> {
+  static OP_CODE = 0x44 // 'D'
+  static OP_NAME = 'OP_DEFI_TX_DESTROY_LOAN_SCHEME'
+
+  composers (dls: DestroyLoanScheme): BufferComposer[] {
+    return [
+      ComposableBuffer.varUIntUtf8BE(() => dls.identifier, v => dls.identifier = v),
+      {
+        fromBuffer: (buffer: SmartBuffer): void => {
+          const num = readBigNumberUInt64(buffer)
+          if (num.isGreaterThan(new BigNumber('0x00000000'))) {
+            dls.height = num
+          }
```

Comment:
In that case, it won't be bi-directionnal with `height: new BigNumber(0)`. See https://github.com/DeFiCh/jellyfish/pull/594/files#r701697406

Either we don't accept height as optional or we lose the bi-directionnal requirement

Reply:
```suggestion
```

Reply:
@Jouzo 
so the reason for the above to change requests is that, in `defid`, it always populates with default value `0` for height.
so it's mandatory in the buffer. so if you consider from that context, I think the bi-directionality is fine.
also if we don't make `height` optional in `DestroyLoanScheme`, the user always has to mention a height. even if they want to delete in the next available block. that would be troublesome.

please comment if you see it otherwise.

Reply:
```
const destroyLoanScheme: DestroyLoanScheme = {
    identifier: 'scheme1',
}
```
toBuffer() will have `0x00000000` in the buffer for height. and fromBuffer() will populate the following
```
{
    identifier: 'scheme1',
    height: new BigNumber(0)
}
```

Reply:
I think both makes sense. 
either we lose the optional property and keep the bi-directional aspect or  inversely we have an easier usage with the optional property but lose the bi-direction. 
I don’t have the insight yet to appreciate this. I’ll follow your decision @surangap

Reply:
ok, taking a step back for the consistency.
we have discussed a similar situation here https://github.com/DeFiCh/jellyfish/pull/608#discussion_r693363296 extensively. And finally agreed to remove the optionality. This makes it not tallying exactly with the RPC interface. But makes things simpler/symmetrical.

I'm thinking if we could add another layer before the transaction builder, that tallies with defid side RPC layer. That will obviously reduce the work for tx crafter. But we can consider it later. For now, tx crafter has to give `height = new BigNumber(0)`, if he want's the `height` optional scenario.